src/shogun/lib/SGVector.h

```diff
@@ -204,6 +206,36 @@ template<class T> class SGVector : public SGReferencedData
 		/** Fill vector with zeros */
 		void zero();
 
+        template <typename U = T> requires Floating<U> && RngVectorizable<U>
+        void random() {
+            auto r = std::make_unique<CRandom>();
+            r->fill_array_co(vector, vlen);
+        }
+
+        template <typename U = T> requires Integral<U> && RngVectorizable<U>
```

Comment:
This is definitely more readable!

Reply:
Yeah, I find it a very nice advantage for programmer happiness. Great stuff for shorter and more clear compiler error messages as well.